libs/wingsdk/src/shared/convert.ts

```diff
@@ -1,52 +1,37 @@
-import { Construct } from "constructs";
+import { createHash } from "crypto";
 import { normalPath } from "./misc";
-import { IInflightHost, IResource, Node, Resource } from "../std";
+import { IInflight } from "../std";
 
 /**
- * Convert a resource with a single method into a resource with a different
- * single method. This is useful for converting between types like
+ * Transform an inflight to one that is wrapped with another one.
+ * The new inflight can have a different signature.
+ * This is useful for converting between types like
  * IFunctionHandler and IQueueSetConsumerHandler.
- *
- * Both the input and return values of this function are expected to be
- * resources with a single method named "handle".
  */
 export function convertBetweenHandlers(
-  scope: Construct,
-  id: string,
-  baseHandler: IResource,
+  baseHandler: IInflight,
   newHandlerClientPath: string,
   newHandlerClientClassName: string,
   args: Record<string, unknown> = {}
-): IResource {
-  class NewHandler extends Resource {
-    private readonly handler: IResource;
-    private readonly args: Record<string, unknown> = {};
+): IInflight {
+  const handlerClient = baseHandler._toInflight();
+  const newCode = `\
+new (require("${normalPath(
+    newHandlerClientPath
+  )}")).${newHandlerClientClassName}({ handler: ${handlerClient}, args: ${JSON.stringify(
+    args
+  )} })`;
 
-    constructor(theScope: Construct, theId: string, handler: IResource) {
-      super(theScope, theId);
-      this.handler = handler;
-      Node.of(this).hidden = true;
-      this.args = args;
-    }
+  const newHandler = {
+    ...baseHandler,
+    _hash: createHash("md5").update(newCode).digest("hex"),
```

Comment:
do we still need the hash?

Reply:
Yeah, there are several resources that need a way to identify inflights so that they can be deduped. (e.g. API is able to reuse cloud.Functions across routes)

Reply:
ok